### PR TITLE
Update document for testing kafka without broker

### PR DESCRIPTION
**Pull request by cuichenli**
two main changes:
1. add the 	`@Connector("smallrye-in-memory")` annotation. otherwise it will report error 
```
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.smallrye.reactive.messaging.memory.InMemoryConnector and qualifiers [@Default]
	- java member: me.cuichenli.BaristaTest#connector
	- declared on CLASS bean [types=[me.cuichenli.BaristaTest, java.lang.Object], qualifiers=[@Default, @Any], target=me.cuichenli.BaristaTest]
	The following beans match by type, but none have matching qualifiers:
		- Bean [class=io.smallrye.reactive.messaging.memory.InMemoryConnector, qualifiers=[@Any, @Connector("smallrye-in-memory")]]
```

The reproducible repository: https://github.com/cuichenli/quarkus-35051-kafka-test-without-broker
2. add test example for testing batch consumer with inmemoryconnector

one minor change:
1. specified the `await` is imported from `awaitability`

### CODE REVIEW

File: docs/src/main/asciidoc/kafka.adoc [hunk 1]
```diff
@@ -2052,7 +2053,75 @@ class BaristaTest {
 <3> Retrieve the outgoing channel (`beverages`) - the channel must have been switched to in-memory in the test resource.
 <4> Use the `send` method to send a message to the `orders` channel.
 The application will process this message and send a message to `beverages` channel.
-<5> Use the `received` method on `beverages` channel to check the messages produced by the application.
+<5> Use the `received` method on `beverages` channel to check the messages produced by the application. The `await` is imported from https://github.com/awaitility/awaitility[awaitility].
+
+If your Kafka consumer is batch based, you will need to send a batch of messages to the channel as per discussed in this https://github.com/smallrye/smallrye-reactive-messaging/discussions/1691#discussioncomment-2471033[Github Discussion].
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka.adoc:2058]: I don't think we should link to a discussion. Our doc should be authoritative.

File: docs/src/main/asciidoc/kafka.adoc [hunk 2]
```diff
@@ -2052,7 +2053,75 @@ class BaristaTest {
 <3> Retrieve the outgoing channel (`beverages`) - the channel must have been switched to in-memory in the test resource.
 <4> Use the `send` method to send a message to the `orders` channel.
 The application will process this message and send a message to `beverages` channel.
-<5> Use the `received` method on `beverages` channel to check the messages produced by the application.
+<5> Use the `received` method on `beverages` channel to check the messages produced by the application. The `await` is imported from https://github.com/awaitility/awaitility[awaitility].
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka.adoc:2056]: I think we should add the imports to the examples instead. That's why we try to do elsewhere as it can be hard to get it right if you don't have the imports.

Note that this is orthogonal to your PR but maybe you could take care of it in a separate commit, instead of adding this sentence?

**Review by gsmet — COMMENTED** [review]
Thanks for taking the time to contribute!

I added a few small comments about the form.

I will let @ozangunalp review the content.

**Comment by cuichenli** [comment]
thanks @gsmet for the review. i will update it accordingly but later - i am currently on something else.

File: docs/src/main/asciidoc/kafka.adoc (hunk 1 above)
Comment (cuichenli) [reply docs/src/main/asciidoc/kafka.adoc]: done

File: docs/src/main/asciidoc/kafka.adoc (hunk 2 above)
Comment (cuichenli) [reply docs/src/main/asciidoc/kafka.adoc]: done

**Comment by cuichenli** [comment]
Hi @gsmet  and @ozangunalp, it has been a while since the last time the pr got reviewed, any chance you can have a look on it when you got a chance? Thanks!

**Comment by gsmet** [comment]
I rebased and squashed. @ozangunalp could you have a look, please? It looks like a worthy addition.

File: docs/src/main/asciidoc/kafka.adoc [hunk 3]
```diff
@@ -2054,6 +2057,76 @@ class BaristaTest {
...
+
+For instance:
+
+[source, java]
+----
+@ApplicationScoped
+public class BeverageProcessor {
+
+    @Incoming("orders")
+    @Outgoing("beverages")
```
Comment (ozangunalp) [inline docs/src/main/asciidoc/kafka.adoc:2070]: The processing method doesn't return any outgoing data. You need to remove this `Outgoing` annotation and the beverages sink in the test code.

**Review by ozangunalp — CHANGES_REQUESTED** [review]
Good addition indeed! The test case needs some tweaking but it'll be nice to add it in the Kafka docs.

File: docs/src/main/asciidoc/kafka.adoc (hunk 3 above)
Comment (cuichenli) [reply docs/src/main/asciidoc/kafka.adoc]: updated

**Review by ozangunalp — APPROVED** [review]
LGTM! Thanks

**Comment by cuichenli** [comment]
> @cuichenli could you squash commits to only have one?

sure, updated!
